corehq/blobs/__init__.py

```diff
@@ -48,3 +48,47 @@ class BlobInfo(namedtuple("BlobInfo", ["identifier", "length", "digest"])):
     def md5_hash(self):
         if self.digest and self.digest.startswith("md5-"):
             return self.digest[4:]
+
+
+class CODES:
+    """Blob type codes.
+
+    A unique blob type code should be assigned to each new area of HQ
+    that will have blobs associated with it. This is mainly intended for
+    analysis purposes (how much blob storage is used per type code?),
+    although it is also useful when debugging to trace a blob identifier
+    back to its parent.
+
+    When adding codes for new models, always use a unique code that has
+    never been used before, preferably one more than the highest
+    existing code. Once a type code has been used it should never be
+    reused for another purpose.
+
+    Each type code associated with couch documents should only reference
+    document types living in a single couch database or SQL models with
+    non-overlapping primary keys. `MetaDB.get` and related methods as
+    well as tools like the `check_blob_logs` management command will not
+    function properly if this contract is broken.
+    """
+    _default = 0        # legacy, do not use
+
+    tempfile = 1
+
+    # SQL forms + XFormInstance, XFormArchived, XFormDeprecated,
+    # XFormDuplicate, XFormInstance-Deleted
+    form = 2
```

Comment:
nit: would be clearer if this was `form_xml`

Reply:
will rename in a follow-up PR

Reply:
https://github.com/dimagi/commcare-hq/pull/21605